crates/ruff_linter/src/rules/flake8_simplify/rules/ast_with.rs

```diff
@@ -168,7 +168,7 @@ pub(crate) fn multiple_with_statements(
             TextRange::new(with_stmt.start(), colon.end()),
         );
         if !checker
-            .indexer()
+            .program()
             .comment_ranges()
```

Comment:
I think this is a case where it could make sense to simply expose `comment_ranges` on `checker` so that it matters less where the comment ranges are actually stored. I don't think it is necessary to change now but it would e.g. help if we move comment ranges in the future again. 

Overall, I think this is a weakness of our `checker` API. Lint rules need to know too much about individual representations. I think it would be better if there's a single `context` and the rules could simply ask:

* Give me the comment ranges
* give me the indexer
* give me the line index 


etc. without having to do deep drilling.

Reply:
Yeah, I completely agree with your sentiment.